registry/interface.go

```diff
@@ -9,15 +9,20 @@ import (
 
 // SignatureRepository provides a storage for signatures
 type SignatureRepository interface {
-	// Lookup finds all signatures for the specified manifest
-	Lookup(ctx context.Context, manifestDigest digest.Digest) ([]digest.Digest, error)
+	// ListSignatureManifests returns all signature manifests given the manifest digest
+	ListSignatureManifests(ctx context.Context, manifestDigest digest.Digest) ([]SignatureManifest, error)
 
 	// Get downloads the signature by the specified digest
 	Get(ctx context.Context, signatureDigest digest.Digest) ([]byte, error)
```

Comment:
nit: I would reword the comment and the method arguments since this can be used to retrieve any content, not just the signature blob.